src/parser/whamm_parser.rs

```diff
@@ -217,6 +216,149 @@ pub fn process_pair(whamm: &mut Whamm, script_count: usize, pair: Pair<Rule>, er
             trace!("Exiting probe_def");
         }
         Rule::EOI => {}
+        //HAS NOT BEEN CHECKED - make sure this is how the parser works (is .next() correct?)
+        //Rule looks like this: fn_def = { ID ~ "(" ~ ( TYPE ~ ID ) ? ~ ("," ~ TYPE ~ ID )* ~ ")" ~ ("->") ~ TYPE ~ block }
+        Rule::fn_def => {
+            trace!("Entering fn_def");
+            let mut pair = pair.into_inner();
+            let fn_name_rule = pair.next().unwrap();
+            let fn_name_line_col = LineColLocation::from(fn_name_rule.as_span());
+            let mut args = vec![];
+            //get every parameter in the function, stopping at arrow - FIX THIS BUG
```

Comment:
Need to add the location here - bugs out when I try to use the col location of p or inner_p